core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -203,6 +205,16 @@ private ClusterTestDefaults getClusterTestDefaults(Class<?> testClass) {
             .orElseGet(() -> EmptyClass.class.getDeclaredAnnotation(ClusterTestDefaults.class));
     }
 
+    private static void applyConfig(ClusterConfig config, ClusterConfigProperty property) {
+        if (property.id() == -1) {
+            config.serverProperties().put(property.key(), property.value());
+        } else if (property.id() >= CONTROLLER_ID_OFFSET) {
```

Comment:
We need to document it to make sure developers are aware of broker/controller id when they try to define per broker/controller configs